src/renderer/actions/queries.ts

```diff
@@ -274,9 +294,9 @@ export function cancelQuery(queryId) {
   };
 }
 
-function stringifyResultToCSV(origRows, delimiter) {
+function stringifyResultToCSV(origRows: [], delimiter: string): Promise<string> {
   if (!origRows.length) {
-    return '';
+    return Promise.resolve('');
```

Comment:
Issue found: Wasn't returning a promise when there are no rows while the caller expect it to always get a promise back.